core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -206,6 +206,21 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
     finalizeTask = Option(task)
   }
 
+  // Set the threshold to 1 billion which leads to an 128MB bitmap and
+  // the actual size of `HighlyCompressedMapStatus` can be much larger than 128MB.
+  // This may crash the driver with an OOM error.
+  private val SHUFFLE_BLOCK_NUMBER_WARNING_THRESHOLD: Long = 1L << 30
+  private val numberOfShuffleBlocks = numPartitions.toLong * partitioner.numPartitions.toLong
+
+  if (numberOfShuffleBlocks > SHUFFLE_BLOCK_NUMBER_WARNING_THRESHOLD) {
+    logWarning(
+      s"The number of shuffle blocks (${numberOfShuffleBlocks}) for shuffleId ${shuffleId} " +
+        s"for ${_rdd} with ${numPartitions} partitions is possibly too large, " +
+        "which could cause the driver to crash with an out-of-memory error. " +
+        "Consider decreasing the number of partitions in this shuffle stage."
+    )
+  }
+
```

Comment:
We can avoid the instance variables here.
Move `SHUFFLE_BLOCK_NUMBER_WARNING_THRESHOLD` to the companion object, and

```suggestion
  {
    val numberOfShuffleBlocks = numPartitions.toLong * partitioner.numPartitions.toLong

    if (numberOfShuffleBlocks > SHUFFLE_BLOCK_NUMBER_WARNING_THRESHOLD) {
      logWarning(
        s"The number of shuffle blocks (${numberOfShuffleBlocks}) for shuffleId ${shuffleId} " +
          s"for ${_rdd} with ${numPartitions} partitions is possibly too large, " +
          "which could cause the driver to crash with an out-of-memory error. " +
          "Consider decreasing the number of partitions in this shuffle stage."
      )
    }
  }
```

Reply:
I would like to suggest not using the companion object for just one constant - we can just inline it in the if condition, we already have a comment explaining the meaning.

Reply:
Feel free to inline - I dont want a new instance variable for this.